cogs/gameplay.py

```diff
@@ -13,46 +13,38 @@ def __init__(self, bot: commands.Bot) -> None:
 
     @commands.command(aliases=["r"])
     @commands.cooldown(1, 5, commands.BucketType.user)
-    async def roll(self, ctx: commands.Context):
+    async def roll(self, ctx: commands.Context, *, tier: str = None):
         """Rolls a set of photocards for claiming."""
         user = await func.get_user(ctx.author.id)
+        actived_potions = {} if tier else func.get_potions(user.get("actived_potions", {}), iufi.POTIONS_BASE)
+
+        query = {}
+        if not tier:
+            query["$set"] = {"cooldown.roll": time.time() + (func.COOLDOWN_BASE["roll"] * (1 - actived_potions.get("speed", 0)))}
+
+        else:
+            tier = tier.lower()
+            tier = func.match_string(tier, iufi.TIERS_BASE.keys())
```

Comment:
for "qr c", will it match it with common or celestial? so for common, we will have to do "qr co", and for celestial we will have to do "qr ce" to avoid confusions?

Reply:
Using ‘qr c’ will result in a match for ‘common’. Indeed, to roll ‘celestial’, we need to use ‘qr ce’ instead of ‘co’.